storage/internal/benchmarks/memory_metric.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// startRecordingMemory writes memory stats to w every interval, until the
+// returned function is called.
+func startRecordingMemory(w io.Writer, oType outputType, interval time.Duration) (cancel func()) {
+	var memStats *runtime.MemStats = &runtime.MemStats{}
+	done := make(chan bool)
+
+	if oType == outputCSV {
+		writeMemoryHeader(w)
+	}
+
+	go (func() {
+		ticker := time.NewTicker(interval)
+
+		for {
+			select {
+			case <-done:
+				return
+			case <-ticker.C:
+				runtime.ReadMemStats(memStats)
```

Comment:
Do you have any sense of the overhead of doing this? Curious whether it would actually affect results at all if you are calling this frequently.